core/server/api_container/server/startosis_engine/kurtosis_starlark_framework/kurtosis_base_builtin_internal.go

```diff
@@ -75,11 +78,24 @@ func printWarningForArguments(argumentSet *builtin_argument.ArgumentValuesSet, b
 		// print if arguments for this builtIn is deprecated.
 		for _, argument := range arguments {
 			if argumentSet.IsSet(argument.Name) && argument.IsDeprecated() {
-				warningMessage := getFormattedWarningMessageForArgument(argument.Deprecation, builtin.Name, argument.Name)
-				starlark_warning.PrintOnceAtTheEndOfExecutionf("%v %v", starlark_warning.WarningConstant, warningMessage)
+				shouldShownDeprecationNotice := true
```

Comment:
Wait, you're initializing it to true, so if `GetMaybeShouldShownDeprecationNoticeFunc()` returns nil, are we going to print something?

Reply:
yes, the current behavior is to print always, and I didn't want to change that. I've just added a new case where the "Deprecation" could be defined for argument values, so could be the case that the received argument value is not deprecated so we don't show the msg.

Reply:
I'll rename the function's name to clarify the behaviour